testsuite/generate-format/tests/staged/diem.yaml

```diff
@@ -168,7 +168,7 @@ ScriptFunction:
           TYPENAME: TypeTag
     - args:
         SEQ:
-          TYPENAME: TransactionArgument
+          SEQ: U8
```

Comment:
IIUC this would technically be a breaking change right?

Reply:
The nice thing here is that since `ScriptFunction` is new there's nothing to break. This could make it harder for clients that are hand-rolling their own transaction builders to update from `Script` to `ScriptFunction`, but I don't think that that should be much of an issue?

Reply:
I guess, the idea is that it hasn't been rolled out yet?

Still I wonder how users feel about BCS-encoded arguments for the new scripts.

Reply:
That's a good question @matbd, what's the rational for this change? At the end of the day the transaction is encoded as bcs so why do we need to have nested encodings like this?

Reply:
I think before, clients had to understand both how to BCS encode the arguments and how to pack them into `TransactionArgument`. Now, they only need to understand the former, so this should hopefully make life simpler.

Reply:
I'm having a hard time seeing how this is "simpler" since I need to do bcs encoding at two different steps (1 for the arguments here and another time when I go to serialize the whole txn)

Reply:
> The nice thing here is that since ScriptFunction is new there's nothing to break.

That is a good point and does answer my initial question about being a breaking change

Reply:
Fair point. My thinking was that because clients already need to know how to do BCS serialization, reusing that knowledge would be easier asking them to understand how to use the `TransactionArgument` format... But nested encoding isn't very nice either.

Reply:
Another reason favoring `Vec<u8>` over `TransactionArgument` might be genericity? With `TransactionArgument`, we can only encode U8, U64, U128, Bool, Address, and U8Vector, however, with BCS-bytes, we can additionally encode, in theory, any Move type, including Struct, Vector of any valid types, etc.